inspirehep/modules/records/receivers.py

```diff
@@ -103,6 +103,30 @@ def assign_uuid(sender, record, *args, **kwargs):
 # models_committed
 #
 
+@models_committed.connect
+def notify_legacy(sender, changes):
+    """Notify Legacy that a given record has been inserted or updated.
+
+    Notifies Legacy that a given record has been inserted or updated, by
+    sharing this information via Redis. Then Legacy will fetch the actual
+    record through subsequent HTTP requests.
+    """
+    redis_connection = None
+
+    def _get_redis_connection():
+        if redis_connection:
+            return redis_connection
+        redis_url = current_app.config.get('CACHE_REDIS_URL')
+        return StrictRedis.from_url(redis_url)
+
+    for model_instance, dummy in changes:
+        if isinstance(model_instance, RecordMetadata):
+            recid = model_instance.json.get('control_number')
+            if recid:
+                redis_connection = _get_redis_connection()
+                redis_connection.sadd('records_to_sync_into_legacy', recid)
```

Comment:
Actually: there is currently no API that given just the recid, it will give back a record. One also needs to know the document type.

E.g.: https://qa.inspirehep.net/api/record/1405870 currently returns `404` while https://qa.inspirehep.net/api/iterature/1405870 correctly works.

So either we make the 404 to actually be a redirect, or we send to legacy not only the recid but also literature.

For the time being legacy will assume that the the set element in redis will contain `1234,literature` (i.e. comma-separated integer and document type).